### PR TITLE
chore: add support for Eclipse CI (Jenkins)

### PR DESCRIPTION
**Pull request by javierron**
Signed-off-by: Javier Ron <javierron90@gmail.com>

### CODE REVIEW

**Comment by javierron** [comment]
@monperrus I managed to scan the repository branches and pull requests from Jenkins, but I'm still unable to trigger a pipeline, I believe the problem is [this](https://issues.jenkins.io/browse/JENKINS-54126?page=com.atlassian.jira.plugin.system.issuetabpanels%3Aall-tabpanel) bug, however, I cannot be sure because I don't have access the plugins panel on Jenkins, and I don't have a way to find out which version of the GitHub plugins we have. Do you have permission to manage the plugins on Jenkins? If not, who should we ask?

**Comment by monperrus** [comment]
Thanks a lot for the progress.

> I don't have a way to find out which version of the GitHub plugins we have

You can now this throw the Jenkins API `https://ci.eclipse.org/repairnator/pluginManager/api/xml?depth=1&xpath=/*/*/shortName|/*/*/version&wrapper=plugins` and HTTP basic auth.
`github-branch-source` is 2.9.1, indeed not the fixed version of the Jira issue you refer to.

```
xml_pp < plugins.xml  | egrep -i -A 1 "github|pull|pipeline"
  <shortName>github-api</shortName>
  <version>1.116</version>
--
  <shortName>github</shortName>
  <version>1.32.0</version>
--
  <shortName>pipeline-build-step</shortName>
  <version>2.13</version>
  <shortName>pipeline-stage-step</shortName>
  <version>2.5</version>
--
  <shortName>pipeline-input-step</shortName>
  <version>2.12</version>
  <shortName>pipeline-graph-analysis</shortName>
  <version>1.10</version>
  <shortName>pipeline-rest-api</shortName>
  <version>2.18</version>
--
  <shortName>pipeline-stage-view</shortName>
  <version>2.18</version>
--
  <shortName>github-branch-source</shortName>
  <version>2.9.1</version>
  <shortName>github-organization-folder</shortName>
  <version>1.6</version>
--
  <shortName>pipeline-model-definition</shortName>
  <version>1.7.2</version>
--
  <shortName>pipeline-milestone-step</shortName>
  <version>1.3.1</version>
--
  <shortName>pipeline-stage-tags-metadata</shortName>
  <version>1.7.2</version>
--
  <shortName>pipeline-model-api</shortName>
  <version>1.7.2</version>
  <shortName>pipeline-model-extensions</shortName>
  <version>1.7.2</version>
--
  <shortName>pipeline-model-declarative-agent</shortName>
  <version>1.1.1</version>
  <shortName>pipeline-github-lib</shortName>
  <version>1.0</version>
--
  <shortName>github-issues</shortName>
  <version>1.2.4</version>
````

> Do you have permission to manage the plugins on Jenkins? 

No

> If not, who should we ask?

Create an issue on https://bugs.eclipse.org/

**Comment by javierron** [comment]
Thanks @monperrus!
 
I have now triggered building and testing. but there are a few things still to do:
- Handle repairnator-core dependency to Build/Test the other modules.
- Setup GitHub webhooks to trigger builds automatically (I still do not have access to the settings panel).
- Figure out deploy stage:
  - We cannot build the repairnator-pipeline docker image in Jenkins, according to the docs.
  - Secrets handling
- Rename CI files and directory `.travis` to `.ci` o similar.

**Comment by monperrus** [comment]
Thanks.

> Setup GitHub webhooks to trigger builds automatically (I still do not have access to the settings panel).

Done. added a hook to `https://ci.eclipse.org/repairnator/github-webhook/`

We may want to fly with this version, for some PRs, even if CD is not implemented. WDYT?

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  agent {
+    kubernetes {
+      label 'my-agent-pod'
+      yaml """
+apiVersion: v1
+kind: Pod
+spec:
+  containers:
+  - name: maven
+    image: javierron/ci-env:latest
```
Comment (monperrus) [inline Jenkinsfile:11]: will be changed to repairnator/ci-env

**Comment by monperrus** [comment]
Almost all green; but seems stuck in Maven-repair, see https://ci.eclipse.org/repairnator/job/repairnator-ci/job/PR-1161/

Let's disable the tests for maven-repair?

**Comment by monperrus** [comment]
@andre15silva FYI once this is merge, we'll rebase all you opened PRs to re-trigger CI.

**Comment by javierron** [comment]
@monperrus @andre15silva

Ok, disabled tests for maven-repair, and updated CI image name to repairnator/ci-env. It should be ready for merge in a couple of minutes.

**Comment by monperrus** [comment]
Everything green. undraft and merge?

**Comment by javierron** [comment]
@monperrus I forgot to undraft :sweat_smile:, ready to merge now.

**Comment by monperrus** [comment]
Thanks a lot!

**Comment by monperrus** [comment]
@andre15silva could you rebase your PRs? Thanks!

**Comment by monperrus** [comment]
Added subtasks in #1152 

* remove skipTests in pipeline CI
* remove skipTests in maven-repair CI
* add continuous deployment to Maven Central
* add continuous deployment to Dockerhub

**Pull request merged**